formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -162,9 +163,7 @@ internal class FormulaManagerImpl<Input, State : Any, Output>(
         val compositeKey = constructKey(formula, input)
         val manager = children
             .findOrInit(compositeKey) {
-                val childTransitionListener = TransitionListener { effects, isValid ->
-                    handleTransition(Transition(effects = effects), !isValid)
```

Comment:
Previously we initialized a new `Transition` when passing side-effects up. Depending on the number of layers, there could be a decent amount of unnecessary allocations.